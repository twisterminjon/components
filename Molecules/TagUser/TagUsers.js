import React from 'react'
import PropTypes from 'prop-types'

import TagUser from './TagUser'

TagUsers.propTypes = {
  /** List of users to display */
  users: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      displayName: PropTypes.string.isRequired,
      profileImage: PropTypes.string,
    })
  ),

  /** Called after a user is clicked */
  onRemove: PropTypes.func.isRequired,

  /** ClassName for the wrapper */
  className: PropTypes.string,

  /** Style for the wrapper */
  style: PropTypes.object,
}

TagUsers.defaultProps = {
  className: '',
  style: {},
}

export default function TagUsers({ users, onRemove, className, style }) {
  return (
    <div className={`tagusers ${className}`.trim()} style={style} data-testid="user-list">
      {users.map(user => (
        <TagUser
          key={user.id}
          displayName={user.displayName}
          profileImage={user.profileImage}
          onClick={() => onRemove(user.id)}
          className="tagusers-tag"
        />
      ))}
    </div>
  )
}
