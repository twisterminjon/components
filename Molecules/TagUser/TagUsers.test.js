import React from 'react'

import TagUsers from './TagUsers'
import users from '../../../Mocks/Users.mock'

describe('TextAreaInput', () => {
  const mockFun = jest.fn()

  it('matches the snapshot', () => {
    const wrapper = window.shallow(<TagUsers onRemove={mockFun} users={users} />)
    expect(wrapper).toMatchSnapshot()
  })
})
