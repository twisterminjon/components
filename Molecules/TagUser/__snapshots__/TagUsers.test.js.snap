// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaInput matches the snapshot 1`] = `
<div
  className="tagusers"
  data-testid="user-list"
  style={Object {}}
>
  <TagUser
    className="tagusers-tag"
    displayName="Agnes Rogers, MD"
    key="3"
    onClick={[Function]}
    profileImage="https://www.fillmurray.com/200/200"
    style={Object {}}
  />
  <TagUser
    className="tagusers-tag"
    displayName="David McGinnis, MD"
    key="9"
    onClick={[Function]}
    profileImage="https://www.fillmurray.com/400/400"
    style={Object {}}
  />
  <TagUser
    className="tagusers-tag"
    displayName="David Smith, MD"
    key="4"
    onClick={[Function]}
    profileImage="https://www.fillmurray.com/600/600"
    style={Object {}}
  />
  <TagUser
    className="tagusers-tag"
    displayName="Jennifer Reyes"
    key="8"
    onClick={[Function]}
    profileImage="https://www.fillmurray.com/500/500"
    style={Object {}}
  />
  <TagUser
    className="tagusers-tag"
    displayName="Paul Jason, RN"
    key="6"
    onClick={[Function]}
    profileImage="https://www.fillmurray.com/700/700"
    style={Object {}}
  />
  <TagUser
    className="tagusers-tag"
    displayName="Rosalyn Stratton, RN"
    key="5"
    onClick={[Function]}
    profileImage="https://www.fillmurray.com/800/800"
    style={Object {}}
  />
  <TagUser
    className="tagusers-tag"
    displayName="Sherry Jackson, MD"
    key="10"
    onClick={[Function]}
    profileImage="https://www.fillmurray.com/200/200"
    style={Object {}}
  />
  <TagUser
    className="tagusers-tag"
    displayName="Tony Hunter"
    key="7"
    onClick={[Function]}
    profileImage="https://www.fillmurray.com/450/450"
    style={Object {}}
  />
</div>
`;
